src/main/java/net/fabricmc/loader/transformer/accessWidener/AccessWidener.java

```diff
@@ -0,0 +1,355 @@
+				throw new UnsupportedOperationException("Unsupported type " + split[1]);
+			}
+		}
+
+		Set<String> parentClasses = new LinkedHashSet<>();
+
+		//Also transform all parent classes
+		for (String clazz : targets) {
+			while (clazz.contains("$")) {
+				clazz = clazz.substring(0, clazz.lastIndexOf("$"));
+				parentClasses.add(clazz);
+			}
+		}
+
+		classes.addAll(targets);
+		classes.addAll(parentClasses);
+	}
+
+	void addOrMerge(Map<EntryTriple, Access> map, EntryTriple entry, String access, Access defaultAccess) {
+		if (entry == null || access == null) {
+			throw new RuntimeException("Input entry or access is null");
+		}
+
+		map.put(entry, applyAccess(access, map.getOrDefault(entry, defaultAccess), entry));
+	}
+
+	private Access applyAccess(String input, Access access, EntryTriple entryTriple) {
+		switch (input.toLowerCase(Locale.ROOT)) {
+		case "accessible":
+			makeClassAccessible(entryTriple);
+			return access.makeAccessible();
+		case "extendable":
+			makeClassExtendable(entryTriple);
+			return access.makeExtendable();
+		case "mutable":
+			return access.makeMutable();
+		default:
+			throw new UnsupportedOperationException("Unknown access type:" + input);
+		}
+	}
+
+	private void makeClassAccessible(EntryTriple entryTriple) {
+		if (entryTriple == null) return;
+		classAccess.put(entryTriple.getOwner(), applyAccess("accessible", classAccess.getOrDefault(entryTriple.getOwner(), ClassAccess.DEFAULT), null));
+	}
+
+	private void makeClassExtendable(EntryTriple entryTriple) {
+		if (entryTriple == null) return;
+		classAccess.put(entryTriple.getOwner(), applyAccess("extendable", classAccess.getOrDefault(entryTriple.getOwner(), ClassAccess.DEFAULT), null));
+	}
+
+	public Access getClassAccess(String className) {
+		return classAccess.getOrDefault(className, ClassAccess.DEFAULT);
+	}
+
+	public Access getFieldAccess(EntryTriple entryTriple) {
+		return fieldAccess.getOrDefault(entryTriple, FieldAccess.DEFAULT);
+	}
+
+	public Access getMethodAccess(EntryTriple entryTriple) {
+		return methodAccess.getOrDefault(entryTriple, MethodAccess.DEFAULT);
+	}
+
+	public Set<String> getTargets() {
+		return classes;
+	}
+
+	private static int makePublic(int i) {
+		return (i & ~(Opcodes.ACC_PRIVATE | Opcodes.ACC_PROTECTED)) | Opcodes.ACC_PUBLIC;
+	}
+
+	private static int makeProtected(int i) {
+		if ((i & Opcodes.ACC_PUBLIC) != 0) {
+			//Return i if public
+			return i;
+		}
+
+		return (i & ~(Opcodes.ACC_PRIVATE)) | Opcodes.ACC_PROTECTED;
+	}
+
+	private static int makeFinalIfPrivate(int i) {
+		if ((i & Opcodes.ACC_PRIVATE) != 0) {
+			return i | Opcodes.ACC_FINAL;
+		}
+
+		return i;
+	}
+
+	private static int removeFinal(int i) {
+		return i & ~Opcodes.ACC_FINAL;
+	}
+
+	public interface Access {
+		Access makeAccessible();
+
+		Access makeExtendable();
+
+		Access makeMutable();
+
+		int apply(int i);
+	}
+
+	public enum ClassAccess implements Access {
+		DEFAULT(i -> i),
+		ACCESSIBLE(i -> makePublic(makeFinalIfPrivate(i))),
```

Comment:
Hmm, a class cannot be private, so `makeFinalIfPrivate` is pointless
https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.1-200-E.1
(private is not an option in this list)

Reply:
What does `private class` get compiled down to? package private?

Reply:
package private I think, which guarantees the enclosing class' access to it as same-level class
 https://github.com/FabricMC/fabric-loader/pull/206#discussion_r395294506

Reply:
Ok, I think we will need to remove this then.